src/sql-parser/src/parser.rs

```diff
@@ -2618,6 +2627,57 @@ impl<'a> Parser<'a> {
         })
     }
 
+    fn parse_create_webhook_source(
+        &mut self,
+        name: UnresolvedItemName,
+        if_not_exists: bool,
+    ) -> Result<Statement<Raw>, ParserError> {
+        // Consume these keywords, because earlier we only peeked them.
+        self.expect_keywords(&[FROM, WEBHOOK])?;
+
+        self.expect_keywords(&[BODY, FORMAT])?;
+
+        // Note: we don't user `parse_format()` here because we support fewer formats than other
+        // sources.
+        let body_format = if self.parse_keyword(JSON) {
```

Comment:
I would suggest using `parse_format()` and then erroring in planning if you can an incompatible format--you're going to need to do that check anyway and I think it's more sane to do that with an error rather than an `unreachable`.

Reply:
Originally I used `parse_format()` but removed it, because the errors it returned were misleading. For example, if a user ran a query like...

```
CREATE SOURCE foo FROM WEBHOOK BODY FORMAT AVRO
```

...the returned error was that it expected a keyword `USING`, and it would keep returning parsing errors until you provided an accurate AVRO schema. But in the end the user would get a planning error anyways. It's a small thing, but it seemed like a bad user experience of "stumbling" through the parser only to then also get rejected by planning.

Refactored to use `self.expect_one_of_keywords(...)`, which is 100% cleaner, thanks for the tip!

Reply:
Ah, that makes a lot of sense. Thanks for providing the context.